src/vsphere_cpi/spec/unit/cloud/vsphere/network_spec.rb

```diff
@@ -255,6 +256,10 @@ module VSphereCloud
              .with('t1-router-id').and_return([])
           expect(switch_provider).to receive(:get_switch_by_id)
              .with('switch-id').and_return(logical_switch)
+          expect(logical_port).to receive(:id)
```

Comment:
Can we move this to above like 
```
let(:logical_port) { instance_double(NSXT::LogicalPort, id: "12345") }
```
Similarly for `switch_ports` too. 
I am suggesting these as values are not changing for the test file, so , we can declare once at the top and be done with it.